crates/accelerate/src/unitary_synthesis.rs

```diff
@@ -627,8 +627,8 @@ fn get_2q_decomposers_from_target(
     }
 
     let target_basis_set = get_target_basis_set(target, qubits[0]);
-    let available_1q_basis: HashSet<&str> =
-        HashSet::from_iter(target_basis_set.get_bases().map(|basis| basis.as_str()));
+    let available_1q_basis: IndexSet<&str> =
```

Comment:
You should use use the ahash hasher here otherwise this will cause a slowdown on lookups in the hash set.